azurelinuxagent/ga/exthandlers.py

```diff
@@ -476,6 +477,12 @@ def handle_ext_handlers(self, goal_state_id):
 
         depends_on_err_msg = None
         extensions_enabled = conf.get_extensions_enabled()
+        engine = PolicyEngine.get_instance()
```

Comment:
I think all of this logic should be behind conf flag. Let's not initialize policy engine or log anything related to policy unless conf.get_extension_policy_enabled() is True

Reply:
On second thought, it's probably fine to keep this as long as the conf check is the first thing that happens in the engine initialize(), I would wait for other reviews before making this change.